### PR TITLE
Implemented wallet sweeper

### LINKED ISSUE #38: Implement a wallet sweeper
This is related to #29 but a correctly built sweeper can not only import any funds from any seed to deposit into the Nault wallet, but also if using your current Nault wallet as source seed and an external account it could also empty the whole wallet with one single click. Double functionality which I don't think any other wallet has.

I want to build this if no one tells me not to. Will probably allow any custom range of index to check (with limitation) similar to this tool: https://tools.nanos.cc/?tool=sweep

### PR DESCRIPTION
**Pull request by Joohansson**
I would like some help battle testing this @BitDesert . The core processing code is pretty much the same as the one I built for https://tools.nanos.cc/?tool=sweep but using the features of Nault, like the PoW source and minimum receive from the settings. It probably needs some design love but it's working.

An easy method for testing this is to generate a seed here: https://tools.nanos.cc/?tool=seed
Then using the QR on the account field and try sending some nano to different indexes and derivation method from Natrium. Then testing with either seed, private key or mnemonic on the sweeper. This is also a perfect simulation of a ledger passphrase recover if using the BIP39/44.

Solves #38 and #29

### CODE REVIEW

**Review by BitDesert — CHANGES_REQUESTED** [review]
Some things I would change:
- Embed the GUI in a `uk-card` like the send dialog
- Remove the "Destination Account field" as you would use it only with accounts setup in Nault
- "Set Min" is basically a "reset" button, so we might call it that way
- "Set Max" sets currently both Start and End pretty high but 100 apart. Why?
- There is a third "Set Max" button that looks the same but just sets "Max TX" back to 100 (which is the default). So it's functionality is the same as a "Reset" button
- The message "An account was swept and funds transferred to the destination!" does not disappear, maybe include the account name as well
- A spinning animation like a the "Send" page would be cool, so you see that it's "doing something"
- There is no "Success" message when it's done

Would it be better to set the type of the secret in a dropdown? Or just let it fiddle out itself?

**Comment by Joohansson** [comment]
New design
image

File: src/app/components/sweeper/sweeper.component.html [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            </div>
+          </div>
+
+          <div class="uk-width-1-1 narrow-div">
+            <div class="uk-form-horizontal">
+              <div class="uk-margin">
+                <label class="uk-form-label">Local Destination<span uk-icon="icon: info;" uk-tooltip title="Local accounts that can be used as destination for the swept funds. Make sure you are able to unlock your Nault wallet."></span></label>
+                <div class="uk-form-controls">
+                  <select class="uk-select" [(ngModel)]="myAccountModel" (change)="setDestination(myAccountModel)">
+                    <option *ngFor="let account of accounts" [value]="account.id">{{ account.id }}</option>
```
Comment (NiFNi) [inline src/app/components/sweeper/sweeper.component.html]: Shouldn't this use the alias set for the account if there is one?

File: src/app/components/sweeper/sweeper.component.html [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+                  </div>
+                </div>
+              </div>
+            </div>
+          </div>
+
+          <div class="uk-width-1-1 narrow-div">
+            <div class="uk-form-horizontal">
+              <div class="uk-margin">
+                <label class="uk-form-label">Local Destination<span uk-icon="icon: info;" uk-tooltip title="Local accounts that can be used as destination for the swept funds. Make sure you are able to unlock your Nault wallet."></span></label>
```
Comment (NiFNi) [inline src/app/components/sweeper/sweeper.component.html]: Shouldn't the label be assiciated with the input field via the for attribute? Same applies to the other labels.
See here for more information: https://stackoverflow.com/questions/12894169/what-is-the-html-for-attribute-in-label

File: src/app/services/util.service.ts [hunk 3]
```diff
@@ -192,6 +214,17 @@ function stringToUint5(string) {
   return uint5;
 }
 
+function isNumeric(val) {
+  //numerics and last character is not a dot and number of dots is 0 or 1
+  let isnum = /^-?\d*\.?\d*$/.test(val) && val != ''
+  if (isnum && String(val).slice(-1) !== '.') {
```
Comment (NiFNi) [inline src/app/services/util.service.ts]: can be simplified to 
```
return isnum && String(val).slice(-1) !== '.'
```

File: src/app/services/util.service.ts [hunk 4]
```diff
@@ -68,6 +75,15 @@ function hexToUint8(hexValue) {
   return uint8;
 }
 
+// Check if string is hexdecimal
+function isHex(h) {
+  let re = /^[0-9a-fA-F]+$/
+  if (re.test(h)) {
+    return true
```
Comment (NiFNi) [inline src/app/services/util.service.ts]: Can be simplified to 
```
return re.test(h)
```

**Review by BitDesert — APPROVED** [review]
Lookin good 👍

**Review by NiFNi — COMMENTED** [review]
I like the way this works! The UI looks good for me aswell. Some small things came to my mind when looking at the sources. I added those as comments on the files :) Do whatever you want with that :P

File: src/app/services/util.service.ts (hunk 3 above)
Comment (Joohansson) [reply src/app/services/util.service.ts]: Thanks nifni, I will make some changes before I merge this PR

**Pull request merged**